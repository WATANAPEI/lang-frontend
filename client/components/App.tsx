import React from "react";
// @ts-ignore
import HeaderBar from "./HeaderBar.tsx";
// @ts-ignore
import Main from "./Main.tsx";
// @ts-ignore
import Footer from "./Footer.tsx";
// @ts-ignore
import SideBar from "./SideBar.tsx";

export class App extends React.Component<any, any> {
  public render() {
    return (
      <div>
        <HeaderBar text="This is Header" />
        <Main />
        <SideBar text="this is sidebar" />
        <Footer text="This is footer" />
      </div>
    );
  }
}
