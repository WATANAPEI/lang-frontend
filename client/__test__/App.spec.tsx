import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
// @ts-ignore
import App from "../components/App.tsx";

describe("<App />", () => {
  const wrapper = createShallow()(<App />);
  it("has HeaderBar with a title text", () => {
    expect(wrapper.find("HeaderBar").text()).toBeTruthy();
  });
  it("has SideBar with a text", () => {
    expect(wrapper.find("SideBar").text()).toBeTruthy();
  });
  it("has Main component", () => {
    expect(wrapper.find("Main").exists()).toEqual(true);
  });
  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
