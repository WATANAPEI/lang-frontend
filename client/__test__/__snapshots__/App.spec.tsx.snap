// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders properly 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-headerBar-1"
  container={true}
  spacing={2}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <HeaderBar
      text="Lang Project"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-sideBar-2"
    item={true}
    xs={3}
  >
    <SideBar
      text="sidebar"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-main-3"
    item={true}
    xs={9}
  >
    <Main />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
